--- a/psychopy/app/coder/coder.py
+++ b/psychopy/app/coder/coder.py
@@ -1794,7 +1794,7 @@ class CoderFrame(wx.Frame):
             initPath=''
         dlg = wx.FileDialog(
             self, message=_translate("Open file ..."),
-            defaultDir=initPath, style=wx.OPEN
+            defaultDir=initPath, style=wx.FD_OPEN
             )
 
         if dlg.ShowModal() == wx.ID_OK:
@@ -1925,7 +1925,7 @@ class CoderFrame(wx.Frame):
         #open dlg
         dlg = wx.FileDialog(
             self, message=_translate("Save file as ..."), defaultDir=initPath,
-            defaultFile=filename, style=wx.SAVE, wildcard=wildcard)
+            defaultFile=filename, style=wx.FD_SAVE, wildcard=wildcard)
         if dlg.ShowModal() == wx.ID_OK:
             newPath = dlg.GetPath()
             # if the file already exists, query whether it should be overwritten (default = yes)
--- a/psychopy/app/dialogs.py
+++ b/psychopy/app/dialogs.py
@@ -527,7 +527,12 @@ class ListWidget(GlobSizer):
         pass
 if __name__=='__main__':
     if wx.version() < '2.9':
-        app = wx.PySimpleApp()
+        class MySimpleApp(wx.App):
+            def OnInit(self):
+                # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+                self.SetAssertMode(wx.PYAPP_ASSERT_SUPPRESS)
+                return True
+        app = MySimpleApp()
     else:
         app = wx.App(False)
     dlg = wx.Dialog(None)
--- a/psychopy/app/preferencesDlg.py
+++ b/psychopy/app/preferencesDlg.py
@@ -341,7 +341,12 @@ class PrefCtrls:
 if __name__=='__main__':
     import preferences
     if wx.version() < '2.9':
-        app = wx.PySimpleApp()
+        class MySimpleApp(wx.App):
+            def OnInit(self):
+                # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+                self.SetAssertMode(wx.PYAPP_ASSERT_SUPPRESS)
+                return True
+        app = MySimpleApp()
     else:
         app = wx.App(False)
     app.prefs=preferences.Preferences()#don't do this normally - use the existing psychopy.prefs instance
--- a/psychopy/app/_psychopyApp.py
+++ b/psychopy/app/_psychopyApp.py
@@ -88,6 +88,9 @@ class PsychoPyApp(wx.App):
             If set to True then startup wizard won't appear and stdout/stderr
             won't be redirected to the Coder
         """
+        # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+        self.SetAssertMode(wx.PYAPP_ASSERT_SUPPRESS)
+
         self.version=psychopy.__version__
         self.SetAppName('PsychoPy2')
 
--- a/psychopy/gui.py
+++ b/psychopy/gui.py
@@ -269,7 +269,7 @@ def fileSaveDlg(initFilePath="", initFil
             #"shelved files (*.shelf)|*.shelf"
     global app  # avoid recreating for every gui
     app = ensureWxApp()
-    dlg = wx.FileDialog(None,prompt, initFilePath, initFileName, allowed, wx.SAVE)
+    dlg = wx.FileDialog(None,prompt, initFilePath, initFileName, allowed, wx.FD_SAVE)
     if dlg.ShowModal() == OK:
         #get names of images and their directory
         outName = dlg.GetFilename()
@@ -312,7 +312,7 @@ def fileOpenDlg(tryFilePath="",
     global app  # avoid recreating for every gui
     app = ensureWxApp()
     dlg = wx.FileDialog(None, prompt, tryFilePath, tryFileName, allowed,
-                        wx.OPEN|wx.FILE_MUST_EXIST|wx.MULTIPLE)
+                        wx.FD_OPEN|wx.FD_FILE_MUST_EXIST|wx.FD_MULTIPLE)
     if dlg.ShowModal() == OK:
         #get names of images and their directory
         fullPaths = dlg.GetPaths()
--- a/psychopy/iohub/util/dialogs.py
+++ b/psychopy/iohub/util/dialogs.py
@@ -22,7 +22,12 @@ class ioHubDialog(object):
         self.display_index=display_index
         if ioHubDialog.wxapp is None:
             if wx.version() < '2.9':
-                ioHubDialog.wxapp = wx.PySimpleApp()
+                class MySimpleApp(wx.App):
+                    def OnInit(self):
+                        # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+                        self.SetAssertMode(wx.PYAPP_ASSERT_SUPPRESS)
+                        return True
+                ioHubDialog.wxapp = MySimpleApp()
             else:
                 ioHubDialog.wxapp = wx.GetApp()
                 if ioHubDialog.wxapp is None:
@@ -172,11 +177,11 @@ class FileDialog(ioHubDialog):
         dstyle=0
         
         if openFile is True:
-            dstyle=dstyle | wx.OPEN
+            dstyle=dstyle | wx.FD_OPEN
         if allowMultipleSelections is True:
-            dstyle=dstyle | wx.MULTIPLE
+            dstyle=dstyle | wx.FD_MULTIPLE
         if allowChangingDirectories is True:
-            dstyle=dstyle | wx.CHANGE_DIR
+            dstyle=dstyle | wx.FD_CHANGE_DIR
         
         fileTypesCombined=""
         if isinstance(fileTypes,(list,tuple)):    
@@ -199,4 +204,4 @@ class FileDialog(ioHubDialog):
         result=self.dialog.ShowModal()
         selections=self.dialog.GetPaths()
         self.Destroy()
-        return result, selections
\ No newline at end of file
+        return result, selections
--- a/psychopy/monitors/MonitorCenter.py
+++ b/psychopy/monitors/MonitorCenter.py
@@ -1100,6 +1100,8 @@ class GammaDlg(wx.Dialog):
 
 class MonitorCenter(wx.App):
     def OnInit(self):
+        # Suppress WXDEBUG assertions, as happens by default with wx2.8.
+        self.SetAssertMode(wx.PYAPP_ASSERT_SUPPRESS)
         frame = MainFrame(None, _translate('PsychoPy Monitor Center'))
         frame.Show(True)
         self.SetTopWindow(frame)
